### PR TITLE
Add benchmark package

### PR DESCRIPTION
**Pull request by piribes**
Use a hunterized fork.
PR: [benchmark#365](https://github.com/google/benchmark/pull/363)

### CODE REVIEW

File: cmake/projects/benchmark/hunter.cmake [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+include(hunter_pick_scheme)
+include(hunter_cmake_args)
+
+hunter_add_version(
+    PACKAGE_NAME
+    benchmark
+    VERSION
+    "1.1.0-hunter"
+    URL
+    "https://github.com/piribes/benchmark/archive/v1.1.0-hunter.tar.gz"
```
Comment (ruslo) [inline cmake/projects/benchmark/hunter.cmake]: Do you want me to create a fork in [hunter-packages](https://github.com/hunter-packages)?

File: cmake/projects/benchmark/hunter.cmake [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+hunter_cmake_args(
+    benchmark
+    CMAKE_ARGS
+    BENCHMARK_ENABLE_TESTING=OFF
+)
+
+hunter_pick_scheme(DEFAULT url_sha1_cmake)
+hunter_cacheable(benchmark)
+hunter_download(PACKAGE_NAME benchmark PACKAGE_INTERNAL_DEPS_ID 1)
```
Comment (ruslo) [inline cmake/projects/benchmark/hunter.cmake]: We don't need `INTERNAL_DEPS_ID` for CMake-based packages, see details:
* https://docs.hunter.sh/en/latest/reference/user-modules/hunter_download.html

File: cmake/projects/benchmark/hunter.cmake (hunk 1 above)
Comment (piribes) [inline cmake/projects/benchmark/hunter.cmake]: Ok.

File: cmake/projects/benchmark/hunter.cmake (hunk 2 above)
Comment (piribes) [inline cmake/projects/benchmark/hunter.cmake]: Got it.

File: cmake/projects/benchmark/hunter.cmake (hunk 1 above)
Comment (ruslo) [inline cmake/projects/benchmark/hunter.cmake]: Created https://github.com/hunter-packages/benchmark

I've added you to team (check your email). Also please read this:
* https://docs.hunter.sh/en/latest/creating-new/cmake-dependencies.html#maintenance

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1523
* https://travis-ci.org/ingenue/hunter/builds/218438535

**Comment by ruslo** [comment]
> Testing

Some toolchains failed. You can exclude broken toolchains by commenting them out from `.travis.yml`/`appveyor.yml` configs (send pull request to the [pkg.benchmark](https://github.com/ingenue/hunter/tree/pkg.benchmark) branch) or investigate problem and do the fixes.

File: cmake/projects/benchmark/hunter.cmake (hunk 1 above)
Comment (piribes) [inline cmake/projects/benchmark/hunter.cmake]: Now it uses hunter-packages/benchmark.

**Comment by ruslo** [comment]
Depends on https://github.com/ingenue/hunter/pull/57

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1566
* https://travis-ci.org/ingenue/hunter/builds/220564649

**Comment by ruslo** [comment]
> Testing

Some toolchains still not working.

**Comment by piribes** [comment]
I saw; I'm trying to reproduce the CI configuration on my fork to investigate the issue, [without success](https://ci.appveyor.com/project/piribes/hunter/build/1.0.2/job/0rnqdwucon93pxb9#L4935).
Could you point me why my build fails with error "CMake Error: The source directory "C:/projects/hunter/examples/benchmark" does not exist"?
Thank you.

**Comment by ruslo** [comment]
> Could you point me why my build fails with error "CMake Error: The source directory "C:/projects/hunter/examples/benchmark" does not exist"?

I guess because example is not merged (it's in this pull request). So to test your changes you should merge `pkg.benchmark` and your branch.

**Comment by piribes** [comment]
If I've understood correctly, it is not possible to merge ruslo/hunter:master directly into ingene/hunter:pkg.benchmark, isn't it?

Since my piribes/hunter:hunter was forked from ruslo/hunter:master, I cannot merge it into piribes/hunter:pkg.benchmark (that is based on ingenue/hunter:pkg.benchmark).

The only option I see is to cherry-pick the commits needed to run the CI, [as I made](https://ci.appveyor.com/project/piribes/hunter/build/1.0.5).

Do you suggest a different approach for merging a package into its related testing branch?

**Comment by ruslo** [comment]
### Adding package

```
> git checkout master # switch to branch ruslo/hunter:master
> git checkout -b pr.mypackage # create new branch for pull request
> git add ... # add code
> git commit -m 'Add mypackage package'
> git push my-hunter-fork-remote pr.mypackage
```

Now you can open pull request to ruslo/hunter:master from myaccount/hunter:pr.mypackage

### Excluding broken builds

If some tests not passed and need to be excluded you have to create new pull request for ingenue/hunter:pkg.mypackage:
```
> git checkout pkg.mypackage # switch to branch ingenue/hunter:pkg.mypackage
> git checkout -b pr.mypackage.excludes
> git add .travis.yml appveyor.yml # never other files here!
> git commit -m 'Exclude broken'
> git push my-hunter-fork-remote pr.mypackage.excludes
```
Open pull request to ingenue/hunter:pkg.mypackage from myaccount/hunter:pr.mypackage.excludes

### Testing pull request on package

Here we are testing changes from pull request (not merged into master yet) on particular package. We need to fetch changes from pull request and merge them with `.travis.yml`/`appveyor.yml` testing instructions for package:
```
> git checkout pkg.mypackage
> git checkout -b pkg.mypackage.pr-N # test pull request N
> git fetch ruslo-hunter-remote refs/pull/N/head:pr-N
> git merge pr-N
> git push my-hunter-fork-remote pkg.mypackage.pr-N
```

> Do you suggest a different approach for merging a package into its related testing branch?

@piribes take a look at last instructions, except you don't need to fetch your own pr-N, just substitute `git merge pr-N` with the name of your branch with changes you want to test.

**Comment by ruslo** [comment]
Closing inactive pull request.

**Pull request closed without merging**